airbyte-notification/src/main/java/io/airbyte/notification/NotificationClient.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.notification;
+
+import java.io.IOException;
+import java.util.Map;
+
+public interface NotificationClient {
+
+  boolean notifyJobFailure(Map<String, Object> metadata) throws IOException, InterruptedException;
```

Comment:
i think there are problems with this interface. it seems like we are throwing away type safety for no reason. behind this iface it is is expecting specific keys e.g. connection id, source definition name, etc. outside of this interface we make sure that they are set up in the JobNotifier, but then we pass a `Map<String, Object>`. Would it make sense to just put in the interface each of the arguments we need so we don't lose the type safety?

Reply:
Ok, I changed the interface, is that better without passing arguments in a `Map`?

Reply:
yes! much nicer.